cpp/order/Steinhardt.cc

```diff
@@ -0,0 +1,510 @@
+                }
+            } // End loop over particle's bonds
+
+        // Normalize!
+        for (unsigned int k = 0; k < (2*m_l+1); ++k)
+            {
+            const unsigned int index = (2*m_l+1) * i + k;
+            // Adding the Qlm of the particle i itself
+            m_QlmiAve.get()[index] += m_Qlmi.get()[index];
+            m_QlmiAve.get()[index] /= neighborcount;
+            m_QlmAve.get()[k] += m_QlmiAve.get()[index];
+            // Add the norm, which is the complex squared magnitude
+            m_QliAve.get()[i] += norm(m_Qlmi.get()[index]);
+            }
+        m_QliAve.get()[i] *= normalizationfactor;
+        m_QliAve.get()[i] = sqrt(m_QliAve.get()[i]);
+        } // Ends loop over particles i for Qlmi calcs
+    }
+
+void Steinhardt::computeNorm()
+    {
+    memset((void*) m_QliNorm.get(), 0, sizeof(float)*m_Np);
+
+    const float normalizationfactor = 4*M_PI/(2*m_l+1);
+
+    // Average Q_lm over all particles, which was calculated in compute
+    for (unsigned int k = 0; k < (2*m_l+1); ++k)
+        {
+        m_Qlm.get()[k] /= m_Np;
+        }
+
+    for (unsigned int i = 0; i < m_Np; ++i)
+        {
+        for (unsigned int k = 0; k < (2*m_l+1); ++k)
+            {
+            // Add the norm, which is the complex squared magnitude
+            m_QliNorm.get()[i] += norm(m_Qlm.get()[k]);
+            }
+        m_QliNorm.get()[i] *= normalizationfactor;
+        m_QliNorm.get()[i] = sqrt(m_QliNorm.get()[i]);
+        }
+    }
+
+void Steinhardt::computeAveNorm()
+    {
+    memset((void*) m_QliAveNorm.get(), 0, sizeof(float)*m_Np);
+
+    const float normalizationfactor = 4*M_PI/(2*m_l+1);
+
+    // Average Q_lm over all particles, which was calculated in compute
+    for (unsigned int k = 0; k < (2*m_l+1); ++k)
+        {
+        m_QlmAve.get()[k] /= m_Np;
+        }
+
+    for (unsigned int i = 0; i < m_Np; ++i)
+        {
+        for (unsigned int k = 0; k < (2*m_l+1); ++k)
+            {
+            // Add the norm, which is the complex squared magnitude
+            m_QliAveNorm.get()[i] += norm(m_QlmAve.get()[k]);
+            }
+        m_QliAveNorm.get()[i] *= normalizationfactor;
+        m_QliAveNorm.get()[i] = sqrt(m_QliAveNorm.get()[i]);
+        }
+    }
+
+void Steinhardt::computeWl()
+    {
+    memset((void*) m_Wli.get(), 0, sizeof(complex<float>)*m_Np);
+
+    // This normalization happens in the Ql calculation but
+    // not for Wl, so we need to undo it. In that calculation
+    // the quantity is multiplied by the normalization factor
+    // and then the result is square rooted, so here we just
+    // divide by the square root.
+    float normalizationfactor = sqrt(4*M_PI/(2*m_l+1));
+
+    // Get wigner3j coefficients from wigner3j.cc
+    m_wigner3jvalues = getWigner3j(m_l);
+
+    for (unsigned int i = 0; i < m_Np; i++)
+        {
+        // Revert Ql normalization
+        m_Qli.get()[i] /= normalizationfactor;
```

Comment:
It doesn't appear like this result is being used - I looked at the original `LocalWl` to confirm. The original code had an option for `enableNormalization` that causes this to be useful, but it doesn't appear that it was ever possible to control this on the Python side. I will remove this in the current changes I am making to the code.

Reply:
^ done.